packages/slate-react/src/components/node.js

```diff
@@ -135,25 +135,28 @@ class Node extends React.Component {
     const decs = decorations.concat(node.getDecorations(editor))
     const childrenDecorations = getChildrenDecorations(node, decs)
 
-    let children = []
-
-    node.nodes.forEach((child, i) => {
-      const isChildSelected = !!indexes && indexes.start <= i && i < indexes.end
-
-      children.push(
-        this.renderNode(child, isChildSelected, childrenDecorations[i])
-      )
-    })
-
     // Attributes that the developer must mix into the element in their
     // custom node renderer component.
     const attributes = { 'data-key': node.key }
 
-    // If it's a block node with inline children, add the proper `dir` attribute
-    // for text direction.
-    if (node.object == 'block' && node.nodes.first().object != 'block') {
```

Comment:
Should have mentioned, I was getting decorator errors without that (on ctrl+z in an empty editor)

Reply:
I was also getting errors in my decorate function on Ctrl+Z in an empty editor, so I included it for that reason. I didn't test _not_ wrapping the decorator code with `if (node.nodes.size) {  }` though, so it may not be necessary. Even if it's not, it made a bit more sense to me looking at the code if it was laid out that way.

Reply:
It doesn't seem to be necessary in a limited test I did.

Reply:
It changes the logic though, so it should be reverted. Instead just using `isLeafBlock` in that one spot should fix it. And if not, we can locate the specific other place to fix instead of wrapping it all in an `if`.

Reply:
Ha yes it definitely makes more sense now that I know of `isLeafBlock`'s existence and it's done now.

Is the comment there strictly correct though? I'm a bit confused by it, because wouldn't this also catch blocks with `text` children as well, not just `inline`?

Reply:
Whoops, let me just put everything back in original order....